includes/reader-revenue/class-stripe-connection.php

```diff
@@ -380,14 +439,17 @@ public static function get_donation_prices() {
 			$stripe        = self::get_stripe_client();
 			$all_prices    = $stripe->prices->all()['data'];
 			$prices_mapped = [];
+			$payment_data  = self::get_stripe_data();
+
 			foreach ( $all_prices as $price ) {
 				$has_metadata_field = isset( $price['metadata'][ self::STRIPE_DONATION_PRICE_METADATA ] );
 				if ( $has_metadata_field ) {
 					$metadata_field_value = $price['metadata'][ self::STRIPE_DONATION_PRICE_METADATA ];
-					if ( 'month' === $metadata_field_value && 'month' === $price['recurring']['interval'] ) {
+					$does_currency_match  = strtolower( $payment_data['currency'] ) === strtolower( $price['currency'] );
```

Comment:
(Non-blocking suggestion)

Nitpick, since this is a simple boolean value:

```suggestion
					$currency_matches  = strtolower( $payment_data['currency'] ) === strtolower( $price['currency'] );
```